packages/compiler/test/checker/effective-type.ts

```diff
@@ -0,0 +1,389 @@
+import { ok, strictEqual } from "assert";
+import { DecoratorContext, ModelType, ModelTypeProperty, Type } from "../../core/types.js";
+import { createTestHost, TestHost } from "../../testing/index.js";
+
+describe("compiler: effective type", () => {
+  let testHost: TestHost;
+  let removeFilter: (model: ModelTypeProperty) => boolean;
+
+  beforeEach(async () => {
+    const removeSymbol = Symbol("remove");
+    testHost = await createTestHost();
+    testHost.addJsFile("remove.js", {
+      $remove: function ({ program }: DecoratorContext, entity: Type) {
+        program.stateSet(removeSymbol).add(entity);
+      },
+    });
+    removeFilter = function (property: ModelTypeProperty) {
+      return !testHost.program.stateSet(removeSymbol).has(property);
+    };
+  });
+
+  it("spread", async () => {
+    testHost.addCadlFile(
+      "main.cadl",
+      `
+      @test model Source {
+        prop: string;
+      }
+      
+      @test model Test {
+        prop: { ...Source };
+      }
+      `
+    );
+    const { Source, Test } = await testHost.compile("./");
+    ok(Source.kind === "Model" && Test.kind === "Model", "expected models");
+
+    const propType = Test.properties.get("prop")?.type as ModelType;
+    const effective = testHost.program.checker.getEffectiveModelType(propType);
+    strictEqual(effective, Source);
+  });
+
+  it("indirect spread", async () => {
+    testHost.addCadlFile(
+      "main.cadl",
+      `
+      @test model Source {
+        prop: string;
+      }
+
+      // Alias here as a named model is its own effective type.
+      alias Spread = { 
+        ...Source
+      };
+
+      @test model Test {
+        test: {...Source}
```

Comment:
Is this supposed to be `...Spread` so that you verify that the type of `test` comes back as `Source` and not another type?

Reply:
yes! Good catch